packages/lambdas/src/document-upload.ts

```diff
@@ -0,0 +1,135 @@
+const api = axios.create();
+const region = getEnvOrFail("AWS_REGION");
+const s3 = makeS3Client(region);
+
+const buildResponse = (status: number, body?: unknown) => ({
+  statusCode: status,
+  body,
+});
+
+type FileData = {
+  mimetype?: string;
+  size?: number;
+  docId: string;
+  cxId: string;
+  patientId: string;
+};
+
+export const handler = async (event: S3Event) => {
+  if (event.Records[0]) {
+    const sourceBucket = event.Records[0].s3.bucket.name;
+    const sourceKey = decodeURIComponent(event.Records[0].s3.object.key.replace(/\+/g, " "));
+    const destinationBucket = "devs.metriport.com";
+    const { destinationKey, queryParams } = getDestinationKeyAndQueryParams(sourceKey);
+
+    const params = {
+      CopySource: encodeURI(`${sourceBucket}/${sourceKey}`),
+      Bucket: destinationBucket,
+      Key: destinationKey,
+    };
+    // Make a copy of the file to the general medical documents bucket
+    try {
+      await s3.copyObject(params).promise();
+    } catch (err) {
+      console.log("Error copying file to general medical documents bucket", JSON.stringify(err));
+      capture.error("Error copying file to general medical documents bucket", {
+        extra: { context: `document-upload`, err, destinationKey, sourceKey },
+      });
+      throw err;
+    }
+
+    // Get file info from the copied file
+    const { size, contentType } = await getFileInfoFromS3(destinationKey, destinationBucket);
+    const fileData = {
+      size,
+      mimetype: contentType,
+      ...getIdsFromKey(destinationKey),
+    };
+    console.log("Got file data:", fileData);
+
+    try {
+      // POST /internal/docs/doc-ref
+      return forwardCallToServer(fileData, queryParams);
+    } catch (err) {
+      console.log("Failed to process uploaded file", JSON.stringify(err));
+      capture.error("Failed to process uploaded file", {
+        extra: { context: `document-upload`, err, fileData },
+      });
+    }
+  }
+};
+
+function removeSuffix(key: string, arg1: string) {
+  const newFileName = key.indexOf(arg1) > -1 ? key.replace(arg1, "") : key;
+  return newFileName;
+}
+
+function getIdsFromKey(destinationKey: string): { cxId: string; patientId: string; docId: string } {
+  if (destinationKey.includes("_")) {
+    const keyParts = destinationKey.split("_");
+    if (keyParts[0] && keyParts[1] && keyParts[2] && keyParts[0].includes("/")) {
+      const cxIdParts = keyParts[0].split("/");
+      if (cxIdParts[0]) {
+        const cxId = cxIdParts[0];
+        const patientId = keyParts[1];
+        const docId = keyParts[2];
+        const fileData = {
+          cxId,
+          patientId,
+          docId,
+        };
+        return fileData;
+      }
+    }
+  }
+  capture.error("Invalid destination key", {
+    extra: { context: `document-upload.ids-from-key`, destinationKey },
+  });
+  throw new MetriportError("Invalid destination key", null, { destinationKey });
+}
```

Comment:
I think this should be concentrated on `packages/core` along side the function that creates the file name... those are intrinsically related:
- from params(cxId, patientId) to name
- from name to params(cxId, patientId)

Reply:
Oh, good idea!